weather/src/applet-read-data.c

```diff
@@ -359,7 +359,7 @@ static guchar *_hmac_sha1_base64(const gchar *key, const gchar *data)
 	size_t key_len = strlen(++key);
 	size_t data_len = strlen(data);
 	uint8_t md[64];  // SHA1_Message_Block_Size = 64 bytes
-	uint md_len=64;
+	guint md_len=64;
```

Comment:
Thank you for reporting this issue with a patch!

Indeed, good catch, `uint` is not really a proper type to use here.

The signature of `HMAC()` seems to be:

```
 unsigned char *HMAC(const EVP_MD *evp_md, const void *key,
               int key_len, const unsigned char *d, int n,
               unsigned char *md, unsigned int *md_len);
```

I guess we can then use `guint` as we already use `guchar`, `gchar` and others.